app/recipe/tests/test_recipe.py

```diff
@@ -1,88 +1,140 @@
 from django.test import TestCase
-from recipe.serializers import RecipeSerializer
-
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
 from recipe.models import Recipe, Ingredient
 
+RECIPES_URL = reverse('recipe-list')
+
 
-class RecipeTests(TestCase):
+class RecipeViewSetTest(TestCase):
+    # def setUp(self):
+    #     self.client = APIClient()
+    # if uncomment above lines then
+    # response = self.client.post(RECIPES_URL, payload, content_type="application/json")  doesn't work, why???
+    # some serialization stuff involved?
 
-    def test_recipe_creation(self):
-        # Mock request with the payload
+    def test_list_recipes(self):
+        response = self.client.get(RECIPES_URL)
+        assert response.status_code == status.HTTP_200_OK
+        assert len(response.data) == Recipe.objects.count()
+        # what else to assert?
```

Comment:
So here since you aren't creating any recipes you're testing that the response is empty. Ideally you'd create a few and then make sure they're in the response